src/common/column_matrix.h

```diff
@@ -130,9 +136,9 @@ class ColumnMatrix {
     for (int32_t fid = 0; fid < nfeature; ++fid) {
       if (type_[fid] == kDenseColumn) {
         const size_t ibegin = boundary_[fid].index_begin;
-        uint32_t* begin = &index_[ibegin];
-        uint32_t* end = begin + nrow;
-        std::fill(begin, end, std::numeric_limits<uint32_t>::max());
+        uint8_t* begin = &index_[ibegin * type_size_];
```

Comment:
It will be more readable if replace by:
```cpp
index_ + ibegin * type_size_;
```
Also, using constructions like common::Span can also help

Reply:
index_ was replaced on vector